cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -0,0 +1,208 @@
+              + clusterSchemaRequest.getTopicName()
+              + "-value/versions";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(
+              suffixUrl, clusterSchemaRequest.getProtocol(), KafkaClustersType.SCHEMA_REGISTRY);
+
+      Map<String, String> params = new HashMap<>();
+      params.put("schema", clusterSchemaRequest.getFullSchema());
+
+      HttpHeaders headers = new HttpHeaders();
+      headers.set("Content-Type", SCHEMA_REGISTRY_CONTENT_TYPE);
+      HttpEntity<Map<String, String>> request = new HttpEntity<>(params, headers);
+      ResponseEntity<String> responseNew =
+          reqDetails.getRight().postForEntity(reqDetails.getLeft(), request, String.class);
+
+      String updateTopicReqStatus = responseNew.getBody();
+      log.info(responseNew.getBody());
+
+      return ApiResponse.builder().result(updateTopicReqStatus).build();
+    } catch (Exception e) {
+      log.error(e.getMessage());
+      if (((HttpClientErrorException.Conflict) e).getStatusCode().value() == 409) {
+        return ApiResponse.builder()
+            .result("Schema being registered is incompatible with an earlier schema")
+            .build();
+      }
+      return ApiResponse.builder().result("Failure in registering schema.").build();
+    }
+  }
+
+  public Map<Integer, Map<String, Object>> getSchema(
+      String environmentVal, KafkaSupportedProtocol protocol, String topicName) {
+    try {
+      log.info("Into getSchema request {} {} {}", topicName, environmentVal, protocol);
+      if (environmentVal == null) {
+        return null;
+      }
+
+      List<Integer> versionsList = getSchemaVersions(environmentVal, topicName, protocol);
+      String schemaCompatibility = getSchemaCompatibility(environmentVal, topicName, protocol);
+      Map<Integer, Map<String, Object>> allSchemaObjects = new TreeMap<>();
+
+      if (versionsList != null) {
+        for (Integer schemaVersion : versionsList) {
+          String suffixUrl =
+              environmentVal + "/subjects/" + topicName + "-value/versions/" + schemaVersion;
+          Pair<String, RestTemplate> reqDetails =
+              clusterApiUtils.getRequestDetails(
+                  suffixUrl, protocol, KafkaClustersType.SCHEMA_REGISTRY);
+
+          Map<String, String> params = new HashMap<>();
+
+          ResponseEntity<HashMap> responseNew =
+              reqDetails.getRight().getForEntity(reqDetails.getLeft(), HashMap.class, params);
+          Map<String, Object> schemaResponse = responseNew.getBody();
+          if (schemaResponse != null) {
+            schemaResponse.put("compatibility", schemaCompatibility);
+          }
+
+          log.info(Objects.requireNonNull(responseNew.getBody()).toString());
+          allSchemaObjects.put(schemaVersion, schemaResponse);
+        }
+      }
+
+      return allSchemaObjects;
+    } catch (Exception e) {
+      log.error("Error from getSchema : " + e.getMessage());
```

Comment:
```suggestion
      log.error("Error from getSchema : ", e);
```
it's better to log the whole trace